RadarSDK/RadarTripOptions.m

```diff
@@ -0,0 +1,82 @@
+//
+//  RadarTripOptions.m
+//  RadarSDK
+//
+//  Copyright © 2020 Radar Labs, Inc. All rights reserved.
+//
+
+#import "RadarTripOptions.h"
+
+@implementation RadarTripOptions
+
+NSString *const kExternalId = @"externalId";
+NSString *const kMetadata = @"metadata";
+NSString *const kDestinationGeofenceTag = @"destinationGeofenceTag";
+NSString *const kDestinationGeofenceExternalId = @"destinationGeofenceExternalId";
+NSString *const kMode = @"mode";
+
+- (instancetype)initWithExternalId:(NSString *_Nonnull)externalId {
+    self = [super init];
+    if (self) {
+        _externalId = externalId;
+        _mode = RadarRouteModeCar;
+    }
+    return self;
+}
+
++ (RadarTripOptions *)tripOptionsFromDictionary:(NSDictionary *)dict {
+    if (!dict) {
+        return nil;
+    }
+
+    RadarTripOptions *options = [[RadarTripOptions alloc] initWithExternalId:dict[kExternalId]];
+    options.metadata = dict[kMetadata];
+    options.destinationGeofenceTag = dict[kDestinationGeofenceTag];
+    options.destinationGeofenceExternalId = dict[kDestinationGeofenceExternalId];
+    NSString *modeStr = dict[kMode];
+    if ([modeStr isEqualToString:@"foot"]) {
+        options.mode = RadarRouteModeFoot;
+    } else if ([modeStr isEqualToString:@"bike"]) {
+        options.mode = RadarRouteModeBike;
+    } else {
+        options.mode = RadarRouteModeCar;
+    }
+    return options;
+}
+
+- (NSDictionary *)dictionaryValue {
+    NSMutableDictionary *dict = [NSMutableDictionary new];
+    dict[kExternalId] = self.externalId;
+    dict[kMetadata] = self.metadata;
+    dict[kDestinationGeofenceTag] = self.destinationGeofenceTag;
+    dict[kDestinationGeofenceExternalId] = self.destinationGeofenceExternalId;
+    dict[kMode] = [Radar stringForMode:self.mode];
+    return dict;
```

Comment:
we may need to have nil guards here as several properties could be nil

Reply:
I think if `tripOptionsFromDictionary` returns `nil` without `externalId` / `mode`, then this function should be fine with just a check at the top for `self == nil`